store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/MiniAccumuloStore.java

```diff
@@ -0,0 +1,274 @@
+import uk.gov.gchq.gaffer.operation.impl.job.GetAllJobDetails;
+import uk.gov.gchq.gaffer.operation.impl.job.GetJobDetails;
+import uk.gov.gchq.gaffer.operation.impl.job.GetJobResults;
+import uk.gov.gchq.gaffer.operation.impl.join.Join;
+import uk.gov.gchq.gaffer.operation.impl.output.ToArray;
+import uk.gov.gchq.gaffer.operation.impl.output.ToCsv;
+import uk.gov.gchq.gaffer.operation.impl.output.ToEntitySeeds;
+import uk.gov.gchq.gaffer.operation.impl.output.ToList;
+import uk.gov.gchq.gaffer.operation.impl.output.ToMap;
+import uk.gov.gchq.gaffer.operation.impl.output.ToSet;
+import uk.gov.gchq.gaffer.operation.impl.output.ToSingletonList;
+import uk.gov.gchq.gaffer.operation.impl.output.ToStream;
+import uk.gov.gchq.gaffer.operation.impl.output.ToVertices;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.operation.GetSchema;
+import uk.gov.gchq.gaffer.store.operation.GetTraits;
+import uk.gov.gchq.gaffer.store.operation.add.AddSchemaToLibrary;
+import uk.gov.gchq.gaffer.store.operation.add.AddStorePropertiesToLibrary;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * An {@link AccumuloStore} that uses an Accumulo {@link MiniAccumuloCluster} to
+ * provide a {@link Connector}.
+ */
+public class MiniAccumuloStore extends AccumuloStore {
+    private static final String BASE_DIRECTORY = "miniAccumuloStoreTest-";
+    private static final String ROOTPW = "rootPW";
+    private MiniAccumuloCluster miniAccumuloCluster = null;
+    private MiniAccumuloConfig miniAccumuloConfig = null;
+    private Connector miniConnector;
+
+    public StoreProperties setUpTestDB(final StoreProperties properties) throws StoreException {
+        setProperties(properties);
+
+        File targetDir = new File("target");
+        File baseDir;
+        if (targetDir.exists() && targetDir.isDirectory()) {
+            baseDir = new File(targetDir, BASE_DIRECTORY + UUID.randomUUID());
+        } else {
+            baseDir = new File(FileUtils.getTempDirectory(), BASE_DIRECTORY + UUID.randomUUID());
+        }
+
+        try {
+            FileUtils.deleteDirectory(baseDir);
+            miniAccumuloConfig = new MiniAccumuloConfig(baseDir, ROOTPW);
+            miniAccumuloConfig.setInstanceName(getProperties().getInstance());
+            miniAccumuloCluster = new MiniAccumuloCluster(miniAccumuloConfig);
+            miniAccumuloCluster.start();
+        } catch (final IOException | InterruptedException e) {
+            throw new StoreException(e.getMessage(), e);
+        }
+
+        // Create the user specified in the properties (if not root)
+        // together with the specified password and give them all authorisations
+        try {
+            if (!getProperties().getUser().equalsIgnoreCase("root")) {
```

Comment:
I don't think Gaffer needs to do this. Certainly it's not done as part of the traditional AccumuloStore implementation. In our [docs](https://gchq.github.io/gaffer-doc/stores/accumulo-store.html#trouble-shooting) we have covered this in the BAD_AUTHORIZATIONS error.

Reply:
This was added because with certain sets of properties we were getting BAD_AUTHORIZATIONS exceptions when they weren't expected.
MockAccumulo has no effective understanding or handling of users.
This code has been added because MiniAccumulo does more properly mimic Accumulo user behaviour.
Root is excluded as it is unnecessary as root automatically always has all permissions and when it is not excluded odd behaviour was encountered on setting the authorisations.